MediaGalleryUi/view/adminhtml/web/js/grid/columns/image.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+define([
+    'jquery',
+    'Magento_Ui/js/grid/columns/image'
+], function ($, image) {
+    'use strict';
+
+    return image.extend({
+        defaults: {
+            bodyTmpl: 'Magento_MediaGalleryUi/grid/columns/image',
+            adobeStockIconUrl: '',
+        },
+
+        /**
+         * Return image name from file name.
+         *
+         * @param {Object} record
+         * @return {String}
+         */
+        getName: function (record) {
+            return this.extractImageName(record.thumbnail_url).replace(/\.[^/.]+$/, "");
+        },
+
+        /**
+         * Return content type of image
+         *
+         * @param {Object} record
+         * @return {String}
+         */
+        getContentType: function (record) {
+            return record['content_type'].replace(/(\image\/)/, "").toUpperCase();;
+        },
+
+        /**
+         * Display source if image was downloaded in adobe stock.
+         *
+         * @param {Object} record
+         * @return {String}
+         */
+        getSource: function (record) {
+            return record.source ? this.adobeStockIconUrl : false
```

Comment:
The source may be not only the Adobe Stock. I think we should simply have a source_icon_url as a grid column. A class on the backend like `\Magento\MediaGalleryUi\Ui\Component\Listing\Columns\Url` may be responsible for preparing such urls

Reply:
ok